### PR TITLE
[Kotlin] basics - replace about.md file with concept files

### PR DESCRIPTION
**Pull request by ErikSchierboom**
In [this issue](https://github.com/exercism/v3/issues/2293) we're describing an evolution of the specification of this repo.
These changes include, amongst others, replacing the contents of the `after.md` document with individual concept documents. There are two documents per concept:

1. An `about.md` file containing the description of the concept
1. A `links.json` file containing a set of useful links related to the concept. 

This looks as follows in the repo:

```
<track>
├── concepts
│   ├── <concept>
│   │   ├── about.md
│   │   └── links.json
│   └── ...
```

This PR applies this change to this exercise by: 
                    
1. Copying the existing `after.md` file's contents to the `about.md` file of each concept listed in the exercise's `concepts` section in the `config.json` file.
1. Extract the links from the `after.md` file and put them into the `links.json` file

Before merging this PR, please check that the contents of the concept documents makes sense.
This is especially true when the exercise unlocks multiple concepts, as then the concept documents should be updated to only contain the information relevant to that concept.

Note: to make reviewing easier, the PR has been split into three separate commits. Please squash this PR upon merging.

### CODE REVIEW

File: languages/kotlin/concepts/basics/about.md [hunk 1]
```diff
@@ -1 +1,41 @@
-TODO: add information on basics concept
+# What's next?
+
+## Type inference
+
+Kotlin compiler can infer types for functions and variables in most of the cases. However, declaring function return types explicitly for **public API** is a good practice.
+
+## Entry point
+
+To run your Kotlin program you need to define so-called entry point: function with name `main` with or without arguments. However, in Exercism course we are using automatic tests that are using other functions defined in exercises.
```
Comment (eparovyshnaya) [inline languages/kotlin/concepts/basics/about.md:9]: Do you think it's worth mentionings, that the `main` must be defined in a script, not in a class, as in Java?

**Review by eparovyshnaya — APPROVED** [review]

File: languages/kotlin/concepts/basics/about.md [hunk 2]
```diff
@@ -1 +1,41 @@
-TODO: add information on basics concept
+# What's next?
+
```
Comment (ErikSchierboom) [inline languages/kotlin/concepts/basics/about.md:2]: This should be removed.

```suggestion
# What's next?

```

File: languages/kotlin/concepts/basics/about.md (hunk 1 above)
Comment (ErikSchierboom) [reply languages/kotlin/concepts/basics/about.md:9]: Yeah that seems wise! Could you suggest a change?

Comment (eparovyshnaya) [reply languages/kotlin/concepts/basics/about.md:9]: _... [top-level function](https://kotlinlang.org/docs/reference/functions.html#function-scope) with name `main` ..._

Fits?

Comment (ErikSchierboom) [reply languages/kotlin/concepts/basics/about.md:9]: Could you perhaps use Github's code suggestion feature? See https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request

This makes it very easy for me to commit changes :)

Comment (eparovyshnaya) [reply languages/kotlin/concepts/basics/about.md:9]: Do not have this interface for unknown reason.

Comment (dector) [inline languages/kotlin/concepts/basics/about.md:9]:
```suggestion
To run your Kotlin program you need to define so-called entry point: top-level function with name `main` with or without arguments. However, in Exercism course we are using automatic tests that are using other functions defined in exercises.
```

Comment (dector) [reply languages/kotlin/concepts/basics/about.md:9]: @eparovyshnaya suggestion syntax is described here (pt. 6):
https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request

It's a bit tricky to discover it indeed :)

File: languages/kotlin/concepts/basics/about.md [hunk 3]
```diff
@@ -1 +1,41 @@
...
+```kotlin
+fun run() {}
+
+// is the same as
+fun run(): Unit { return Unit }
+```
+
+Returning `Unit` (and using it in function declaration) is completely optional and is omitted in most of the cases.
+
+[intellij-idea-ic]: https://www.jetbrains.com/idea/download/
```
Comment (dector) [inline languages/kotlin/concepts/basics/about.md:41]: Did I get it right that we need to extract links like that into `links.json`.

Comment (ErikSchierboom) [reply languages/kotlin/concepts/basics/about.md:41]: Probably. Not every link that is in the `about.md` document needs to be extracted to the `links.json` file. For more information, see https://github.com/exercism/v3/issues/2499.

**Review by dector — APPROVED** [review]

File: languages/kotlin/concepts/basics/about.md [hunk 4]
```diff
@@ -1 +1,41 @@
-TODO: add information on basics concept
+# What's next?
+
+## Type inference
```
Comment (ErikSchierboom) [inline languages/kotlin/concepts/basics/about.md:3]:
```suggestion
## Type inference
```

**Pull request merged**